compiler/rustc_hir_analysis/src/astconv/mod.rs

```diff
@@ -2135,6 +2138,34 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
                     // Find all the `impl`s that `qself_ty` has for any trait that has the
```

Comment:
can you move the whole `if let` error reporting chain into a separate function (and module `astconv/errors.rs`). having this much error reporting in a function with relevant behavior makes it harder to read.